oar/core/notification_mgr.py

```diff
@@ -0,0 +1,112 @@
+import os
+import smtplib
+from email.mime.text import MIMEText
+from email.mime.multipart import MIMEMultipart
+from oar.core.config_store import ConfigStore
+from oar.core.exceptions import NotificationException
+from oar.core.worksheet_mgr import WorksheetManager
+import logging
+
+logger = logging.getLogger(__name__)
+
+class NotificationManager:
+    """
+    NotifiationManager will be used to notificate messages via email or slack.
+    """
+    str1 = ''
+    def __init__(self, cs: ConfigStore):
+        if cs:
+            self._cs = cs
+        else:
+            raise NotificationException("argument config store is required") 
+    
+        self.google_app_passwd = self._cs.get_google_app_passwd()
```

Comment:
- mail client and slack client should be initialized in NM constructor, works as a module level variable. e.g. `self.mc` `self.sc`
- mail and slack client should not be initialized by config store. e.g. `MailClient(email_id, app_pwd)`
- mail auth func should be called when initializing mail client
- get subject/get mail content should be responsibility of `MessageHelper`
- func `send_msg_via_email` name can be refined with `share_new_report_url`, in this func we will do following things: manipulate email message and slack message. send out msg via mail client and slack client.

Reply:
@rioliu-rh Updated according the comments, pls review again, thanks